src/main/java/io/openshift/booster/catalog/BoosterCatalogService.java

```diff
@@ -308,34 +308,47 @@ public Path copy(Booster booster, Path projectRoot) throws IOException
    }
 
    @Override
-   public Set<Mission> getMissions()
+   public Set<Mission> getMissions(String ...labels)
    {
-      return boosters.stream()
+      return filtered(boosters.stream(), labels)
                .map(Booster::getMission)
                .collect(Collectors.toCollection(TreeSet::new));
    }
 
+   private Stream<Booster> filtered(Stream<Booster> stream, final String[] labels) {
+      if( labels == null || labels.length==0 )
+         return stream;
+      return stream.filter(x -> {
+         for (String label : labels) {
+            if( !x.getLabels().contains(label) ) {
```

Comment:
Can it throw NPE if no labels exist in a Booster descriptor?

Reply:
If it's missing, it will just be an empty list.